src/results/site.jl

```diff
@@ -0,0 +1,107 @@
+# *********************************************************************************
+"""
+	add_site_results(m::JuMP.AbstractModel, p::REoptInputs, d::Dict)
+
+Adds the Site results to the dictionary passed back from `run_reopt` using the solved model `m` and the `REoptInputs`.
+
+Site results:
+- `annual_renewable_electricity_kwh`
+- `renewable_electricity_pct`
+- `total_renewable_energy_pct`
+- `year_one_emissions_tCO2`
+- `year_one_emissions_tNOx`
+- `year_one_emissions_tSO2`
+- `year_one_emissions_tPM25`
+- `year_one_emissions_from_fuelburn_tCO2`
+- `year_one_emissions_from_fuelburn_tNOx`
+- `year_one_emissions_from_fuelburn_tSO2`
+- `year_one_emissions_from_fuelburn_tPM25`
+- `lifecycle_emissions_cost_CO2`
+- `lifecycle_emissions_cost_health`
+- `lifecycle_emissions_tCO2`
+- `lifecycle_emissions_tNOx`
+- `lifecycle_emissions_tSO2`
+- `lifecycle_emissions_tPM25`
+- `lifecycle_emissions_from_fuelburn_tCO2`
+- `lifecycle_emissions_from_fuelburn_tNOx`
+- `lifecycle_emissions_from_fuelburn_tSO2`
+- `lifecycle_emissions_from_fuelburn_tPM25`
+
+calculated in combine_results function if BAU scenario is run:
+- `lifecycle_emissions_reduction_CO2_pct`
+
+"""
+function add_site_results(m::JuMP.AbstractModel, p::REoptInputs, d::Dict; _n="")
+	r = Dict{String, Any}()
+
+	# renewable elec
+	r["annual_renewable_electricity_kwh"] = round(value(m[:AnnualREEleckWh]), digits=2)
+	r["renewable_electricity_pct"] = round(value(m[:AnnualREEleckWh])/value(m[:AnnualEleckWh]), digits=6)
+
+	# total renewable 
+	add_re_tot_calcs(m,p)
+	r["total_renewable_energy_pct"] = round(value(m[:AnnualRETotkWh])/value(m[:AnnualTotkWh]), digits=6)
+	
+	# pass through for breakeven cost of CO2 calculation:
+	r["include_climate_in_objective"] = p.s.settings.include_climate_in_objective
+	r["pwf_emissions_cost_CO2_grid"] = p.pwf_emissions_cost["CO2_grid"]
+	r["pwf_emissions_cost_CO2_onsite"] = p.pwf_emissions_cost["CO2_onsite"]
+
+	# Year 1 Emissions results at Site level
+	r["year_one_emissions_tCO2"] = round(value(m[:EmissionsYr1_Total_LbsCO2] * TONNE_PER_LB), digits=2)
+	r["year_one_emissions_tNOx"] = round(value(m[:EmissionsYr1_Total_LbsNOx] * TONNE_PER_LB), digits=2)
+	r["year_one_emissions_tSO2"] = round(value(m[:EmissionsYr1_Total_LbsSO2] * TONNE_PER_LB), digits=2)
+	r["year_one_emissions_tPM25"] = round(value(m[:EmissionsYr1_Total_LbsPM25] * TONNE_PER_LB), digits=2)
+
+	r["year_one_emissions_from_fuelburn_tCO2"] = round(value(m[:yr1_emissions_onsite_fuel_lbs_CO2] * TONNE_PER_LB), digits=2)
+	r["year_one_emissions_from_fuelburn_tNOx"] = round(value(m[:yr1_emissions_onsite_fuel_lbs_NOx] * TONNE_PER_LB), digits=2)
+	r["year_one_emissions_from_fuelburn_tSO2"] = round(value(m[:yr1_emissions_onsite_fuel_lbs_SO2] * TONNE_PER_LB), digits=2)
+	r["year_one_emissions_from_fuelburn_tPM25"] = round(value(m[:yr1_emissions_onsite_fuel_lbs_PM25] * TONNE_PER_LB), digits=2)
+
+	# Lifecycle emissions results at Site level
+	if !isnothing(p.s.site.bau_grid_emissions_lb_CO2_per_year)
```

Comment:
Should this just be `bau_emissions_lb_CO2_per_year` instead of grid emissions? Presumably someone could set the grid emissions to zero and just look at fuel burn emissions.

Reply:
good catch!